.golangci.yaml

```diff
@@ -0,0 +1,173 @@
+run:
+  deadline: 5m
+  skip-dirs:
+    - mock*
+  skip-files:
+  - "zz_generated.*\\.go$"
+  - ".*conversion.*\\.go$"
+linters:
+  disable-all: true
+  enable:
+    - asciicheck
+    - bodyclose
+    - deadcode
+    - depguard
+    - dogsled
+    - errcheck
+    - errorlint
+    - exportloopref
+    - goconst
+    - gocritic
+    - godot
+    - gofmt
+    - goimports
+    - gosimple
+    - govet
+    - importas
+    - gosec
+    - ineffassign
+    - misspell
+    - nakedret
+    - nilerr
+    - noctx 
+    - nolintlint
+    - prealloc
+    - predeclared
+    - revive
+    - rowserrcheck
+    - staticcheck
+    - structcheck
+    - stylecheck
+    - thelper
+    - typecheck
+    - unconvert
+    - unused
+    - varcheck
+    - whitespace
+  # Run with --fast=false for more extensive checks
+  fast: true
+
+linters-settings:
+  gosec:
+    go: "1.17"
+    severity: medium
+    confidence: medium
+    excludes:
+      - G107
+  importas:
+      no-unaliased: true
+      alias:
+        # Kubernetes
+        - pkg: k8s.io/api/core/v1
+          alias: corev1
+        - pkg: k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1
+          alias: apiextensionsv1
+        - pkg: k8s.io/apimachinery/pkg/apis/meta/v1
+          alias: metav1
+        - pkg: k8s.io/apimachinery/pkg/api/errors
+          alias: apierrors
+        - pkg: k8s.io/apimachinery/pkg/util/errors
+          alias: kerrors
+        # Controller Runtime
+        - pkg: sigs.k8s.io/controller-runtime
+          alias: ctrl
+        # KCP
+        - pkg: sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1
+          alias: controlplanev1
+        # CAPI
+        - pkg: sigs.k8s.io/cluster-api/api/v1alpha3
+          alias: clusterv1alpha3
+        - pkg: sigs.k8s.io/cluster-api/api/v1alpha4
+          alias: clusterv1alpha4
+        - pkg: sigs.k8s.io/cluster-api/api/v1beta1
+          alias: clusterv1
+        # CAPM3
+        - pkg: github.com/metal3-io/cluster-api-provider-metal3/api/v1alpha5
+          alias: infrav1alpha5
+        - pkg: github.com/metal3-io/cluster-api-provider-metal3/api/v1beta1
+          alias: infrav1
+        # BMO
+        - pkg: github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1
```

Comment:
have you verified if all these packages are used in the code base?

Reply:
ow now, I have just copy-pasted capm3 one, so I need to scan through go.mod file to check it these exists right ?

Reply:
not go mod better look through of files if all of those packages are imported or not.

Reply:
Done